include/lsst/afw/image/PhotoCalib.h

```diff
@@ -0,0 +1,426 @@
+    double const value;
+    double const sigma;
+};
+
+class PhotoCalib : public table::io::PersistableFacade<PhotoCalib>, public table::io::Persistable {
+public:
+    // no move or copy
+    PhotoCalib(PhotoCalib const &) = delete;
+    PhotoCalib(PhotoCalib &&) = delete;
+    PhotoCalib & operator=(PhotoCalib const &) = delete;
+    PhotoCalib & operator=(PhotoCalib &&) = delete;
+
+    /**
+     * Create a empty, zeroed calibration.
+     */
+    PhotoCalib(): PhotoCalib(0) {}
+
+    /**
+     * Create a non-spatially-varying calibration.
+     *
+     * @param[in]  countsMag0       The constant counts/magnitude zero point (counts at magnitude 0).
+     * @param[in]  countsMag0Sigma  The error on the zero point.
+     * @param[in]  bbox             The bounding box on which this PhotoCalib is valid.
+     */
+    explicit PhotoCalib(double countsMag0, double countsMag0Sigma=0,
+                        afw::geom::Box2I const & bbox=afw::geom::Box2I()):
+        _countsMag0(countsMag0), _countsMag0Sigma(countsMag0Sigma), _isConstant(true) {
+            ndarray::Array<double, 2, 2> coeffs = ndarray::allocate(ndarray::makeVector(1, 1));
+            coeffs[0][0] = countsMag0;
+            _zeroPoint = std::make_shared<afw::math::ChebyshevBoundedField>(
+                                    afw::math::ChebyshevBoundedField(bbox, coeffs));
+        }
+
+    /**
+     * Create a spatially-varying calibration.
+     *
+     * @param[in]  zeroPoint      The spatially varying photometric zero point.
+     * @param[in]  countsMag0Sigma  The error on the zero point.
+     */
+    PhotoCalib(std::shared_ptr<afw::math::BoundedField> zeroPoint, double countsMag0Sigma=0):
+        _zeroPoint(zeroPoint), _countsMag0(computeCountsMag0(zeroPoint)),
+        _countsMag0Sigma(countsMag0Sigma), _isConstant(false) {}
+
+    /**
+     * Create a calibration with a pre-computed mean. Primarily for de-persistence.
+     *
+     * @param[in]  countsMag0      The mean of the zeroPoint() over its bounding box.
+     * @param[in]  zeroPoint       The spatially varying photometric zero point.
+     * @param[in]  countsMag0Sigma The error on the zero point.
+     * @param[in]  isConstant      Is this PhotoCalib spatially constant?
+     */
+    PhotoCalib(double countsMag0, double countsMag0Sigma,
+               std::shared_ptr<afw::math::BoundedField> zeroPoint,
+               bool isConstant):
+        _zeroPoint(zeroPoint), _countsMag0(countsMag0), _countsMag0Sigma(countsMag0Sigma),
+        _isConstant(isConstant) {}
+
+    //@{
+    /**
+     * Convert counts in ADU to maggies.
+     *
+     * If passed point, use the exact calculation at that point, otherwise, use the mean scaling factor.
+     *
+     * @param[in]  counts  The source counts in ADU.
+     * @param[in]  point   The point that counts is measured at (must be within the domain of the
+     *                     BoundedField of this PhotoCalib).
```

Comment:
And as I think about this a bit more, the thing we'd probably most want to protect against (using one `Exposure`'s `PhotoCalib` to convert the fluxes of points from a different `Exposure`) won't be protected by checking a bbox, since both `PhotoCalib`s will have the "same" bbox: `(0,4000)`.